src/rosdistro/manifest_provider/git.py

```diff
@@ -53,35 +53,50 @@ def git_manifest_provider(_dist_name, repo, pkg_name):
         raise RuntimeError('Unable to fetch package.xml: %s' % e)
 
 
+def _git_version_gte(version):
+    global _git_client_version
+    if not _git_client_version:
+        cmd = [_git_client_executable, '--version']
+        result = _run_command(cmd)
+        _git_client_version = result['output'].split()[-1]
+    return _git_client_version.split('.') >= version.split('.')
```

Comment:
String comparison is not adequate here. It needs to compare the integer values. Otherwise "5" > "10".

Also the Python module `distutils.version` contains classes and functions to parse and compare version numbers.